include/aspect/particle/generator/interface.h

```diff
@@ -100,15 +112,15 @@ namespace aspect
            */
           virtual
           void
-          generate_particles(std::multimap<types::LevelInd, Particle<dim> > &particles) = 0;
+          generate_particles(std::multimap<Particles::internal::LevelInd, Particle<dim> > &particles) = 0;
```

Comment:
This suggests a design flaw in the deal.II classes -- if other codes are using internal data structures, we've done something wrong :-(